src/app/item-popup/ItemStat.tsx

```diff
@@ -81,6 +96,16 @@ export default function ItemStat({ stat, item }: { stat: DimStat; item: DimItem
   );
 }
 
+// returns the socket associated with an applied weapon mod
+function modSocketFor(item) {
+  return (
+    item.sockets &&
+    item.sockets.sockets.find((socket) => {
+      return socket.plug && socket.plug.plugItem.itemCategoryHashes.includes(1052191496);
```

Comment:
Because plugItem is a raw inventory item def, you need to handle the case where itemCategoryHashes are null